src/citrine/resources/design_workflow.py

```diff
@@ -16,13 +17,65 @@ class DesignWorkflowCollection(Collection[DesignWorkflow]):
     _individual_key = None
     _collection_key = 'response'
     _resource = DesignWorkflow
+    _api_version = "v2"
 
-    def __init__(self, project_id: UUID, session: Session):
+    def __init__(self, project_id: UUID, session: Session, branch_id: UUID = None):
         self.project_id: UUID = project_id
         self.session: Session = session
+        self.branch_id: UUID = branch_id
+
+    def register(self, model: DesignWorkflow) -> DesignWorkflow:
+        """
+        Upload a new design workflow.
+
+        The model's branch ID is ignored in favor of this collection's. If this
+        collection has a null branch ID, then the model is uploaded to the v1
+        endpoint and retrieved separately, to ensure all details are included.
+
+        Parameters
+        ----------
+        model: DesignWorkflow
+            The design workflow to be uploaded.
+
+        Return
+        -------
+        DesignWorkflow
+            The newly created design workflow.
+
+        """
+        if self.branch_id is None:
+            # To create a design workflow without providing a branch ID, we need to hit the v1
+            # API, then do a GET to grab the ID of the branch that was created automatically.
+            v1 = _DesignWorkflowCollectionV1(self.project_id, self.session)
+            # Passing in the subclass instance as self avoids infinite recursion.
+            created_dw = super(DesignWorkflowCollection, v1).register(model)
+            return super().get(created_dw.uid)
+        else:
+            # branch_id is in the body of design workflow endpoints, so it must be serialized.
+            # This means the collection branch_id might not match the workflow branch_id. The
+            # collection should win out, since the user is explicitly referencing the branch
+            # represented by this collection.
+            # To avoid modifying the parameter, and to ensure the only change is the branch_id, we
+            # deepcopy, modify, then register it.
+            model_copy = deepcopy(model)
```

Comment:
This is an interesting case. Would this ever be intentional? I think this should raise a warning, and possibly even an error.

Reply:
I chatted with James about this, and the conclusion was that this allows a user to create the same design workflow on different branches more conveniently. I don't know how common this will be of course, but it seems reasonable to support.